src/Squirrel/UpdateManager.ApplyReleases.cs

```diff
@@ -35,7 +35,20 @@ public ApplyReleasesImpl(string rootAppDirectory)
 
                 progress(0);
                 status("Applying updates");
-                
+
+                // Try to remove older versions before applying releases
+                try
+                {
+                    var currentVersion = updateInfo.CurrentlyInstalledVersion != null ?
+                        updateInfo.CurrentlyInstalledVersion.Version : null;
+
+                    await removeOldVersionFolders(new SemanticVersion[] { currentVersion });
```

Comment:
Please add a comment, that `currentVersion` is not the one to be deleted but rather the only one to be excluded from the deletion...

Reply:
Exactly.